pirogue_cli/android/device.py

```diff
@@ -46,9 +47,18 @@ def is_rooted(self):
                 shell=True,
                 stderr=subprocess.PIPE, )
             self.has_adb_root = True
-            if 'adbd cannot run as root in production builds' in output.decode():
+            if (
+                'adbd cannot run as root in production builds' in output.decode()
+                # This happens when root is disabled in developer options
+                or 'ADB Root access is disabled' in output.decode()
+            ):
                 self.has_adb_root = False
                 try:
+                    # This small sleep step is required to ensure adb is
+                    # correctly recovering after failed adb root.
+                    # Otherwise, one will get an "error: closed" stderr.
+                    time.sleep(1)
```

Comment:
It's not perfect and 1 is sometimes not enough. 5 does the job for sure, but introduces some latency, not sure which one to decide for.